src/meta/meta_turtle.c

```diff
@@ -135,6 +135,15 @@ __metadata_load_bulk(WT_SESSION_IMPL *session)
 
 err:
     WT_TRET(__wt_metadata_cursor_release(session, &cursor));
+    /*
+     * We want to explicitly close, not just release the metadata cursor here. We know we are in
+     * initialization and this open cursor holds a lock on the metadata and we may need to verify
+     * the metadata.
+     */
+    if (session->meta_cursor != NULL) {
+        WT_TRET(session->meta_cursor->close(session->meta_cursor));
+        session->meta_cursor = NULL;
+    }
```

Comment:
Can we have a metadata cursor close function()?

I was thinking something like this:
```
diff --git a/src/conn/conn_dhandle.c b/src/conn/conn_dhandle.c
index e6da84467..3d4edc581 100644
--- a/src/conn/conn_dhandle.c
+++ b/src/conn/conn_dhandle.c
@@ -797,8 +797,7 @@ restart:
      * it's potentially used when discarding other open data handles. Close it before discarding the
      * underlying metadata handle.
      */
-    if (session->meta_cursor != NULL)
-        WT_TRET(session->meta_cursor->close(session->meta_cursor));
+    WT_TRET(__wt_metadata_cursor_close(session));
 
     /* Close the remaining handles. */
     WT_TAILQ_SAFE_REMOVE_BEGIN(dhandle, &conn->dhqh, q, dhandle_tmp)
diff --git a/src/include/extern.h b/src/include/extern.h
index 93c103e80..a3e714350 100644
--- a/src/include/extern.h
+++ b/src/include/extern.h
@@ -1064,6 +1064,8 @@ extern int __wt_metadata_btree_id_to_uri(WT_SESSION_IMPL *session, uint32_t btre
   WT_GCC_FUNC_DECL_ATTRIBUTE((warn_unused_result));
 extern int __wt_metadata_cursor(WT_SESSION_IMPL *session, WT_CURSOR **cursorp)
   WT_GCC_FUNC_DECL_ATTRIBUTE((warn_unused_result));
+extern int __wt_metadata_cursor_close(WT_SESSION_IMPL *session)
+  WT_GCC_FUNC_DECL_ATTRIBUTE((warn_unused_result));
 extern int __wt_metadata_cursor_open(WT_SESSION_IMPL *session, const char *config,
   WT_CURSOR **cursorp) WT_GCC_FUNC_DECL_ATTRIBUTE((warn_unused_result));
 extern int __wt_metadata_cursor_release(WT_SESSION_IMPL *session, WT_CURSOR **cursorp)
diff --git a/src/meta/meta_table.c b/src/meta/meta_table.c
index a35a39a5b..7a398f62f 100644
--- a/src/meta/meta_table.c
+++ b/src/meta/meta_table.c
@@ -133,6 +133,16 @@ __wt_metadata_cursor(WT_SESSION_IMPL *session, WT_CURSOR **cursorp)
     return (0);
 }
 
+/*
+ * __wt_metadata_cursor_close --
+ *     Close a metadata cursor.
+ */
+int
+__wt_metadata_cursor_close(WT_SESSION_IMPL *session)
+{
+    return (session->meta_cursor == NULL ? 0 : session->meta_cursor->close(session->meta_cursor));
+}
+
 /*
  * __wt_metadata_cursor_release --
  *     Release a metadata cursor.
diff --git a/src/meta/meta_turtle.c b/src/meta/meta_turtle.c
index d232a249a..41d6d2252 100644
--- a/src/meta/meta_turtle.c
+++ b/src/meta/meta_turtle.c
@@ -134,16 +134,13 @@ __metadata_load_bulk(WT_SESSION_IMPL *session)
     WT_ERR_NOTFOUND_OK(ret, false);
 
 err:
-    WT_TRET(__wt_metadata_cursor_release(session, &cursor));
     /*
      * We want to explicitly close, not just release the metadata cursor here. We know we are in
      * initialization and this open cursor holds a lock on the metadata and we may need to verify
      * the metadata.
      */
-    if (session->meta_cursor != NULL) {
-        WT_TRET(session->meta_cursor->close(session->meta_cursor));
-        session->meta_cursor = NULL;
-    }
+    WT_TRET(__wt_metadata_cursor_release(session, &cursor));
+    WT_TRET(__wt_metadata_cursor_close(session));
     return (ret);
 }
```

Reply:
Mine ended up looking like:
```
+int
+__wt_metadata_cursor_close(WT_SESSION_IMPL *session)
+{
+    WT_DECL_RET;
+
+    if (session->meta_cursor != NULL)
+        ret = session->meta_cursor->close(session->meta_cursor);
+    session->meta_cursor = NULL;
+    return (ret);
+}
```